### PR TITLE
feat: Add active state for star button (#750)

### PR DESCRIPTION
**Pull request by pontakornth**
I added the active class and mixin for star icon button.
It's same as #754 but I can't change the branch of pull request so I make a new one.

Issue: #750

### CODE REVIEW

**Review by shahednasser — CHANGES_REQUESTED** [review]
Please run `npm run compile` and commit the changes

**Comment by pontakornth** [comment]
@shahednasser OK. Done it.

**Review by shahednasser — CHANGES_REQUESTED** [review]
I still need you to add the active state button in `assets/js/buttons-examples.js` next to the star button.

File: src/components/icon/_star.less [hunk 1]
```diff
@@ -9,10 +14,11 @@
     font-variant: normal;
     text-rendering: auto;
     -webkit-font-smoothing: antialiased;
+    transition: color 0.25s;
```
Comment (shahednasser) [inline src/components/icon/_star.less:17]: Remove transition as it is making the button look bad. Don't forget to run `npm run compile` afterwards

Comment (pontakornth) [reply src/components/icon/_star.less]: Got it.

**Comment by shahednasser** [comment]
Thank you!

@all-contributors please add @pontakornth for code

**Pull request merged**